source/geometries/Next100Shielding.cc

```diff
@@ -343,123 +550,224 @@ namespace nexus {
+            	  vertex = lat_beam_gen_->GenerateVertex("INSIDE");
+            	}
+            	else if (rand_beam ==1){
+            	  // //lat_2 (lat_beam_x,-beam_thickness_1/2.,-lateral_z_separation_/2.)
+            	  vertex = lat_beam_gen_->GenerateVertex("INSIDE");
+            	  vertex.setZ(vertex.z() - lateral_z_separation_);
+            	}
+            	else if (rand_beam ==2){
+            	  // //lat_3 	(-lat_beam_x,-beam_thickness_1/2.,lateral_z_separation_/2.)
+            	  vertex = lat_beam_gen_->GenerateVertex("INSIDE");
+            	  vertex.setX(vertex.x() - (shield_x_ + 2*steel_thickness_ + lead_thickness_));
+            	}
+            	else if (rand_beam ==3){
+            	  // //lat_4 (-lat_beam_x,-beam_thickness_1/2.,-lateral_z_separation_/2.)
+            	  vertex = lat_beam_gen_->GenerateVertex("INSIDE");
+            	  vertex.setX(vertex.x() - (shield_x_ + 2*steel_thickness_ + lead_thickness_));
+            	  vertex.setZ(vertex.z() - lateral_z_separation_);
+            	}
+            	else if (rand_beam ==4){
+            	  // //lat_5 front_beam (-front_x_separation_/2.,-beam_thickness_1/2.,front_beam_z)
+            	  vertex = front_beam_gen_->GenerateVertex("INSIDE");
+            	}
+            	else if (rand_beam ==5){
+            	  // //lat_6 front_beam (front_x_separation_/2.,-beam_thickness_1/2.,front_beam_z)
+            	  vertex = front_beam_gen_->GenerateVertex("INSIDE");
+            	  vertex.setX(vertex.x() + front_x_separation_);
+            	}
+            	else if (rand_beam ==6){
+            	  // //lat_7 front_beam (-front_x_separation_/2.,-beam_thickness_1/2.,-front_beam_z)
+            	  vertex = front_beam_gen_->GenerateVertex("INSIDE");
+            	  vertex.setZ(vertex.z() - (shield_z_+2*steel_thickness_+lead_thickness_));
+            	}
+            	else if (rand_beam ==7){
+            	  //lat_8 front_beam (front_x_separation_/2.,-beam_thickness_1/2.,-front_beam_z)
+            	  vertex = front_beam_gen_->GenerateVertex("INSIDE");
+            	  vertex.setX(vertex.x() + front_x_separation_);
+            	  vertex.setZ(vertex.z() - (shield_z_+2*steel_thickness_+lead_thickness_));
+            	}
+            }
+        }
+
+    else if(region=="PEDESTAL"){
+        G4double rand = G4UniformRand();
+
+        if (rand < perc_ped_bottom_vol_) { //SUPPORT-BOTTOM
       	  if (G4UniformRand() < 0.5) {
-      	    vertex = front_roof_gen_->GenerateVertex("INSIDE");
-      	    vertex.setZ(vertex.z() + (shield_z_/2.+steel_thickness_+lead_thickness_/2.));
-	    // std::cout<<"frontal +"<<std::endl;
+      	    vertex = ped_support_bottom_gen_->GenerateVertex("INSIDE");
       	  }
       	  else{
-      	    vertex = front_roof_gen_->GenerateVertex("INSIDE");
-      	    vertex.setZ(vertex.z() - (shield_z_/2.+steel_thickness_+lead_thickness_/2.));
-      	    // std::cout<<"frontal -"<<std::endl;
+      	    vertex = ped_support_bottom_gen_->GenerateVertex("INSIDE");
```

Comment:
Instead of generating the vertex twice, couldn't you just use the same vertex and shift it if the random number is less than 0.5? The same for the rest of the code and also in the `SHIELDING_STRUCT` generation. Please, tell me if I'm not seeing something here.